Author: Pino Toscano <pino@debian.org>
Description: Correct determineFileSystemTypeImpl() for any glibc-based system
 Fixes compilation on GNU/kFreeBSD, which has no determineFileSystemTypeImpl()
 being recognized by default.
Last-Update: 2011-12-03
Forwarded: no

--- a/kdecore/io/kfilesystemtype_p.cpp
+++ b/kdecore/io/kfilesystemtype_p.cpp
@@ -53,7 +53,7 @@ KFileSystemType::Type determineFileSyste
     return kde_typeFromName(buf.f_fstypename);
 }
 
-#elif defined(Q_OS_LINUX) || defined(Q_OS_HURD)
+#elif defined(Q_OS_LINUX) || defined(Q_OS_HURD) || defined(__GLIBC__)
 # include <sys/vfs.h>
 # ifdef QT_LINUXBASE
    // LSB 3.2 has statfs in sys/statfs.h, sys/vfs.h is just an empty dummy header
